pkg/identityserver/end_device_registry.go

```diff
@@ -336,6 +438,16 @@ func (is *IdentityServer) deleteEndDevice(ctx context.Context, ids *ttnpb.EndDev
 	return ttnpb.Empty, nil
 }
 
+func validateEndDeviceAuthenticationCode(authCode ttnpb.EndDeviceAuthenticationCode) error {
+	if validFrom, validTo := ttnpb.StdTime(authCode.ValidFrom), ttnpb.StdTime(authCode.ValidTo); validFrom != nil &&
+		validTo != nil {
```

Comment:
Shouldn't this also check whether `validTo` is not in the past?

Reply:
One of the ways of making the CAC invalid for testing is to play around with `validFrom` and `validTo`. I get your point but I don't know if we should check for this. 
Users must be aware of what `validTo` means right?

Reply:
I personally prefer not allowing the creation with a `ValidTo` and `ValidFrom` previous than the time of creation but at the same time it feels like an edge case where users won't end up falling in. Out of curiosity, what exactly is the test case where it would benefit from this? Wouldn't setting the `validTo` and timing out for a few seconds cover the `past valid date` case?

Reply:
> Wouldn't setting the validTo and timing out for a few seconds cover the past valid date case

Yes that would work. But please note that the people who use the stack are not usually engineers and who may not have the intuition to do that. Whereas simply setting something in the past date is intuitive.

Reply:
True enough. Then, considering that users might want to play around with the `validTo` value, this looks appropriate.